modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/AutoDownloadSettingsFragment.kt

```diff
@@ -434,24 +414,17 @@ class AutoDownloadSettingsFragment :
             )
     }
 
+    @SuppressLint("CheckResult")
     private fun setupNewEpisodesToggleStatusCheck() {
-        val value = viewModel.getAutoDownloadNewEpisodes()
-        when (value) {
-            Podcast.AUTO_DOWNLOAD_OFF -> {
-                newEpisodesPreference?.isChecked = false
-            }
-
-            Podcast.AUTO_DOWNLOAD_NEW_EPISODES -> {
-                newEpisodesPreference?.isChecked = true
-            }
-
-            else -> {
-                // This is the case where users have not set this toggle yet.
-                // In this case, we check if the user has auto download enabled for any podcast
-                // so we can enable the global auto-download status.
-                newEpisodesPreference?.isChecked = viewModel.hasEpisodesWithAutoDownloadEnabled.value
-            }
-        }
+        viewModel.countPodcastsAutoDownloading()
+            .map { it > 0 }
+            .subscribeBy(
+                onError = { Timber.e(it) },
+                onSuccess = { on ->
+                    onNewEpisodesToggleChange(on.toAutoDownloadStatus())
+                    newEpisodesPreference?.isChecked = on
+                },
+            )
```

Comment:
I put this code back to fix this issue. https://github.com/Automattic/pocket-casts-android/pull/3020/files